crates/uv-install-wheel/src/wheel.rs

```diff
@@ -459,32 +461,37 @@ fn install_script(
             use std::fs::Permissions;
             use std::os::unix::fs::PermissionsExt;
 
-            let permissions = fs::metadata(&path)?.permissions();
+            // We fall back to copy when the file is on another drive and if we don't own the file.
+            rename_or_copy.rename_or_copy(&path, &script_absolute)?;
 
-            if permissions.mode() & 0o111 == 0o111 {
-                // If the permissions are already executable, we don't need to change them.
-                rename_with_retry_sync(&path, &script_absolute)?;
-            } else {
-                // If we have to modify the permissions, copy the file, since we might not own it.
```

Comment:
I think this change is incorrect... If we have to change the permissions, we _must_ copy the file. Check out the Git history on this.